### PR TITLE
feat: add  workflows for tests and dependabot

### PR DESCRIPTION
**Pull request by JGSweets**
Adds workflows for:
* dependabot
* running flake8
* running pytest

### CODE REVIEW

File: .github/workflows/test-python-package.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        python -m pip install --upgrade pip
+        python -m pip install flake8 pytest pytest-cov
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+        if [ -f requirements-test.txt ]; then pip install -r requirements-test.txt; fi
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 --statistics --count --exit-zero
```
Comment (ssharpe42) [inline .github/workflows/test-python-package.yml:35]: does this error out on the current main?

Comment (JGSweets) [reply .github/workflows/test-python-package.yml:35]: This adds test to run remotely that didn't exist. Now PRs will have tests to validate working code.

Comment (JGSweets) [reply .github/workflows/test-python-package.yml:35]: I don't believe so. I ran a test on a personal repo too to validate the flake8 would pass:
https://github.com/JGSweets/synthetic-data/actions/runs/3101760273/jobs/5023430869
However, tests won't pass until https://github.com/capitalone/synthetic-data/pull/272 bc of the seed issue.

Comment (ssharpe42) [reply .github/workflows/test-python-package.yml:35]: Ok I think merging this in first is good so we have automated tests for the other pr even if they fail currently for main

**Review by ssharpe42 — APPROVED** [review]

**Pull request merged**